lib/dpul_collections/user_sets.ex

```diff
@@ -44,6 +45,40 @@ defmodule DpulCollections.UserSets do
     Repo.all_by(Set, user_id: scope.user.id)
   end
 
+  def list_user_sets_for_addition(%Scope{} = scope, solr_id \\ nil) do
```

Comment:
Can we add spec to this function? It's not clear until you read the test what it's returning, in particular what "has_solr_id" means.